common/firmware/spi/spi_comms.cpp

```diff
@@ -25,12 +26,16 @@ using namespace spi;
  * Public Functions
  */
 
-Spi::Spi() { SPI2_init(); }
+Spi::Spi(SPI_interface SPI_intf_instance) : SPI_intf(SPI_intf_instance) {}
 
 void Spi::transmit_receive(const BufferType& transmit, BufferType& receive) {
-    Reset_CS_Pin();
+    Reset_CS_Pin(SPI_intf.GPIO_handle, SPI_intf.pin);
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
     hal_transmit_receive(const_cast<uint8_t*>(transmit.data()), receive.data(),
-                         BUFFER_SIZE, TIMEOUT);
-    Set_CS_Pin();
+                         BUFFER_SIZE, TIMEOUT,
+                         (SPI_HandleTypeDef*)this->SPI_intf.SPI_handle);
+    /*
```

Comment:
can we remove this kind of comment? it just restates what happens in the line of code.